src/utils/emitter.ts

```diff
@@ -0,0 +1,16 @@
+export interface OpenCloseState {
```

Comment:
@Elijbet Instead of exporting the interface, I was thinking that the `OpenCloseComponent` interface would have a set of required functions like this:

```tsx
export interface OpenCloseComponent {
  /**
   * The host element.
   */
  readonly el: HTMLElement;

 beforeOpen: () => void;
  open: () => void;
  beforeClose: () => void;
  close: () => void;
}
```

What do you think?

Otherwise, we aren't really forcing the developer to implement the interface as we want because the only requirement right now is the `readonly el: HTMLElement;`

Reply:
That makes sense. Thanks for the explanation.